connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala

```diff
@@ -335,43 +340,51 @@ object KustoWriter {
     if (row.get(fieldIndexInRow, dataType) == null) null else dataType match {
       case DateType => DateTimeUtils.toJavaDate(row.getInt(fieldIndexInRow)).toString
       case TimestampType => dateFormat.format(DateTimeUtils.toJavaTimestamp(row.getLong(fieldIndexInRow)))
-      case StringType => GetStringFromUTF8(row, fieldIndexInRow, nested)
+      case StringType => GetStringFromUTF8(row.getUTF8String(fieldIndexInRow), nested)
       case BooleanType => row.getBoolean(fieldIndexInRow).toString
-      case _: StructType => convertStructToCsv(row.getStruct(fieldIndexInRow, dataType.asInstanceOf[StructType].length), dataType.asInstanceOf[StructType], dateFormat)
-      case _: ArrayType => convertArrayToCsv(row.getArray(fieldIndexInRow), dataType, dateFormat)
+      case strType: StructType => convertStructToCsv(row.getStruct(fieldIndexInRow, dataType.asInstanceOf[StructType].length), strType, dateFormat)
```

Comment:
Why strType and not _ ? Don't you get a warning of an unused variable?

Reply:
i use it .. this replaces the calling to asInstanceOf[StructType]